drivers/flash/Kconfig.nor

```diff
@@ -8,7 +8,7 @@ menuconfig SPI_NOR
 	select FLASH_HAS_DRIVER_ENABLED
 	select FLASH_HAS_PAGE_LAYOUT
 	select FLASH_JESD216
-	depends on SPI
+	select SPI
```

Comment:
Likewise with this, as the name implies SPI_NOR is SPI, it cannot reside on an i2c bus